arch/arc/CMakeLists.txt

```diff
@@ -20,3 +20,4 @@ if(CONFIG_ISA_ARCV2)
 endif()
 
 add_subdirectory(core)
+add_subdirectory(arcmwdt)
```

Comment:
This could lead to the impression that `arcmwdt` is always active.

I propose to move the safeguard one level up, so that it is clearer already at this level that arcmwdt is only used with Metaware compiler.
```suggestion
if(COMPILER STREQUAL arcmwdt)
  add_subdirectory(arcmwdt)
endif()
```

Reply:
Thanks for suggestion - now fixed!